apps/settings/js/carrier.js

```diff
@@ -272,463 +169,6 @@ var CarrierSettings = (function() {
-            pendingAutomaticSelectionRequest = true;
-          } else {
-            doEnableAutomaticSelection();
-          }
-        } else {
-          pendingAutomaticSelectionRequest = false;
-          scan();
-        }
-      }
-
-      function getAutomaticSelection(index) {
-        return opAutoSelectStates[index];
-      }
-
-      // API
-      return {
-        stop: stop,
-        scan: scan,
-        setAutomaticSelection: setAutomaticSelection,
-        getAutomaticSelection: getAutomaticSelection
-      };
-    })(document.getElementById('availableOperators'));
-
-    updateSelectionMode(true);
-  }
-
-  /**
-   * Update the checkbox of the automatic operator selection.
-   */
-  function cs_updateAutomaticOperatorSelectionCheckbox() {
-    var opAutoSelectInput =
-      document.querySelector('#operator-autoSelect input');
-    var targetIndex = DsdsSettings.getIccCardIndexForCellAndDataSettings();
-    opAutoSelectInput.checked =
-      gOperatorNetworkList.getAutomaticSelection(targetIndex);
-    opAutoSelectInput.dispatchEvent(new Event('change'));
-  }
-
-  /**
-   * Init roaming preference selector.
-   */
-  function cs_initRoamingPreferenceSelector() {
-    if (!_mobileConnection.getRoamingPreference) {
-      document.getElementById('operator-roaming-preference').hidden = true;
-      return;
-    }
-
-    var defaultRoamingPreferences =
-      Array.prototype.map.call(_mobileConnections,
-        function() { return 'any'; });
-    var roamingPreferenceHelper =
-      SettingsHelper('ril.roaming.preference', defaultRoamingPreferences);
-
-    var selector =
-      document.getElementById('operator-roaming-preference-selector');
-    var req = _mobileConnection.getRoamingPreference();
-    req.onsuccess = function() {
-      for (var i = 0; i < selector.options.length; i++) {
-        var selection = selector.options[i];
-        if (selection.value === req.result) {
-          selection.selected = true;
-
-          var evt = document.createEvent('Event');
-          evt.initEvent('change', true, true);
-          selector.dispatchEvent(evt);
-          break;
-        }
-      }
-    };
-
-    req.onerror = function() {
-      console.warn('carrier: ' + req.error.name);
-    };
-
-    selector.addEventListener('blur', function() {
-      var index = this.selectedIndex;
-      if (index >= 0) {
-        var selection = this.options[index];
-        roamingPreferenceHelper.get(function gotRP(values) {
-          var targetIndex =
-            DsdsSettings.getIccCardIndexForCellAndDataSettings();
-          var setReq = _mobileConnection.setRoamingPreference(selection.value);
-          setReq.onsuccess = function set_rp_success() {
-            values[targetIndex] = selection.value;
-            roamingPreferenceHelper.set(values);
-          };
-          setReq.onerror = function set_rp_error() {
-            selector.value = values[targetIndex];
-          };
-        });
-      }
-    });
-  }
-
```

Comment:
This will be done in another bug. This one is more focusing on the operator settings.